catalog/resource_sql_table.go

```diff
@@ -242,6 +255,39 @@ func (ti *SqlTableInfo) serializeColumnInfos() string {
 	return strings.Join(columnFragments[:], ", ") // id INT NOT NULL, name STRING, age INT
 }
 
+func serializePrimaryKeyConstraint(constraint ConstraintInfo) string {
+	constraint_clause := fmt.Sprintf("CONSTRAINT %s PRIMARY KEY(%s)", constraint.getWrappedConstraintName(), constraint.getWrappedKeyColumnNames())
+	if constraint.Rely {
+		constraint_clause += " RELY"
+	}
+	return constraint_clause
+}
+
+func serializeForeignKeyConstraint(constraint ConstraintInfo) string {
+	constraint_clause := fmt.Sprintf("CONSTRAINT %s FOREIGN KEY(%s) REFERENCES %s", constraint.getWrappedConstraintName(), constraint.getWrappedKeyColumnNames(), constraint.ParentTable)
+	if len(constraint.ParentColumns) > 0 {
+		constraint_clause += fmt.Sprintf("(%s)", constraint.getWrappedParentColumnNames())
+	}
+	if constraint.Rely {
+		constraint_clause += " RELY"
+	}
+	return constraint_clause
+}
+
+func (ti *SqlTableInfo) serializeConstraints() string {
+	constraintFragments := make([]string, len(ti.Constraints))
+
+	for i, constraint := range ti.Constraints {
+		if constraint.Type == "PRIMARY KEY" {
+			constraintFragments[i] = serializePrimaryKeyConstraint(constraint)
+		} else if constraint.Type == "FOREIGN KEY" {
```

Comment:
Since we only support `PRIMARY KEY` and `FOREIGN KEY`, we should fail client side if some other value is used.